pydra/engine/task.py

```diff
@@ -287,6 +288,7 @@ def __init__(self, func: ty.Callable,
         elif 'return' in func.__annotations__:
             raise NotImplementedError('Branch not implemented')
         self.output_spec = output_spec
+        self.set_output_keys()
```

Comment:
After thinking about this some more, it's probably better to make it an `Output` class ala `Inputs`.

Something along the lines of
```python
klass = make_klass(self.output_spec)
self.outputs = klass(**{f.name: (None if f.default is dc.MISSING
                                  else f.default) for f in dc.fields(klass)})
```

This might have to be done at a higher level __init__ (`FunctionTask`, `ShellCommandTask`)

Reply:
`output` is not `result`. and that's why it was removed. there is a property `output_names` that will return the output names. and `get_result()` will return a result object.

Reply:
i see - i'm looking at this from the scope of connecting via Futures


```python
wf = Workflow(...)

N1 = Task(interface=fsl.BET(),
          in_file="somefile",
          ...)

# at this point, N1.outputs should consist of Futures

N2 = Task(interface=fsl.MCFLIRT(),
          in_file=N1.outputs.out_file,
          ...)

wf.run()  # should produce N1 --> N2 --> N2.output
```

but perhaps this could just be changed to `N1.results.out_file`?

Reply:
i would perhaps suggest something like `N1.result.out_file` so this object would be a future.